### PR TITLE
docs: Add up-to-date documentation and remove legacy documentation (DEV-4229)

### PR DESCRIPTION
**Pull request by BalduinLandolt**
This PR:

- removes the old documentation moved over from the old `dsp-meta-svc` repo, which was
  - partially entirely obsolete (examples, RDF-renderings, etc.)
  - partially outdated
- adds some new documentation, which should be up to date, and easier to keep updated in case of future changes

### CODE REVIEW

File: docs/data/datamodel.md [hunk 1]
```diff
@@ -1,8 +1,333 @@
...
+
+| Field     | Type   | Cardinality | Restrictions      |
+| --------- | ------ | ----------- | ----------------- |
+| `__type`  | string | 1           | Literal 'License' |
+| `license` | url    | 1           |                   |
+| `date`    | date   | 1           |                   |
+| `details` | string | 0-1         |                   |
+
+
+#### attribution
```
Comment (seakayone) [inline docs/data/datamodel.md:217]:
```suggestion
#### Attribution
```

File: docs/data/datamodel.md [hunk 2]
```diff
@@ -1,8 +1,333 @@
...
+| `__type`     | string | 1           | Literal 'Address' |
+| `street`     | string | 1           |                   |
+| `postalCode` | string | 1           |                   |
+| `locality`   | string | 1           |                   |
+| `country`    | string | 1           |                   |
+| `canton`     | string | 0-1         |                   |
+| `additional` | string | 0-1         |                   |
+
+
+#### license
```
Comment (seakayone) [inline docs/data/datamodel.md:207]:
```suggestion
#### License
```

File: docs/data/datamodel.md [hunk 3]
```diff
@@ -1,8 +1,333 @@
...
+
+#### publication
+
+| Field  | Type   | Cardinality | Restrictions |
+| ------ | ------ | ----------- | ------------ |
+| `text` | string | 1           |              |
+| `url`  | url    | 0-1         |              |
+
+
+#### address
```
Comment (seakayone) [inline docs/data/datamodel.md:194]:
```suggestion
#### Address
```

File: docs/data/datamodel.md [hunk 4]
```diff
@@ -1,8 +1,333 @@
...
+#### dmp (Data Management Plan)
+
+| Field      | Type    | Cardinality | Restrictions                 |
+| ---------- | ------- | ----------- | ---------------------------- |
+| `__type`   | string  | 1           | Literal 'DataManagementPlan' |
+| `available | boolean | 0-1         |                              |
+| `url`      | url     | 0-1         |                              |
+
+
+#### publication
```
Comment (seakayone) [inline docs/data/datamodel.md:186]:
```suggestion
#### Publication
```

File: docs/data/datamodel.md [hunk 5]
```diff
@@ -1,8 +1,333 @@
...
+
+| Field    | Type   | Cardinality | Restrictions                                                                                                                                |
+| -------- | ------ | ----------- | ------------------------------------------------------------------------------------------------------------------------------------------- |
+| `__type` | string | 1           | Literal 'URL'                                                                                                                               |
+| `type`   | string | 1           | Literal 'URL', 'Geonames', 'Pleiades', 'Skos', 'Periodo', 'Chronontology', 'GND', 'VIAF', 'Grid', 'ORCID', 'Creative Commons', 'DOI', 'ARK' |
+| `url`    | string | 1           |                                                                                                                                             |
+| `text`   | string | 0-1         |                                                                                                                                             |
+
+
+#### dmp (Data Management Plan)
```
Comment (seakayone) [inline docs/data/datamodel.md:177]:
```suggestion
#### Data Management Plan (`dmp`)
```

File: docs/data/datamodel.md [hunk 6]
```diff
@@ -1,8 +1,333 @@
...
+}
+```
+
+
+#### date
+
+String with the format `YYYY-MM-DD`.
+
+
+#### url
```
Comment (seakayone) [inline docs/data/datamodel.md:160]:
```suggestion
#### URL
```

File: docs/data/datamodel.md [hunk 7]
```diff
@@ -1,8 +1,333 @@
...
+
+```json
+{
+    "en": "Lorem ipsum in English.",
+    "de": "Lorem ipsum auf Deutsch."
+}
+```
+
+
+#### date
```
Comment (seakayone) [inline docs/data/datamodel.md:155]:
```suggestion
#### Date
```

File: docs/data/datamodel.md [hunk 8]
```diff
@@ -1,8 +1,333 @@
...
+| `__type`  | string | 1           | Literal 'Grant'            |
+| `funders` | id[]   | 1-n         | Person or Organization IDs |
+| `number`  | string | 0-1         |                            |
+| `name`    | string | 0-1         |                            |
+| `url`     | url    | 0-1         |                            |
+
+
+### Value Types
+
+#### lang_string
```
Comment (seakayone) [inline docs/data/datamodel.md:143]:
```suggestion
#### Language Tag (`lang_string')
```

**Review by seakayone — APPROVED** [review]

File: docs/data/datamodel.md [hunk 9]
```diff
@@ -1,8 +1,333 @@
...
+
+Object with a language tag as key and a string as value.
```
Comment (mpro7) [inline docs/data/datamodel.md:145]:
```suggestion
Object with a language ISO code as key and a string as value.
```

File: docs/data/datamodel.md [hunk 10]
```diff
@@ -1,8 +1,333 @@
...
+
+String with the format `YYYY-MM-DD`.
+
+
+#### url
+
+An object representing a URL. 
+Depending on the `type` field,
+the URL may be a generic URL
+or a more specific link, like P PID
```
Comment (mpro7) [inline docs/data/datamodel.md:165]:
```suggestion
or a more specific link, like PID
```

File: docs/data/datamodel.md [hunk 11]
```diff
@@ -1,8 +1,333 @@
...
+This includes thoughts on mappings to external aggregators 
+and on RDF representation of the metadata.
+
+
+## Issues/Questions
+
+- access conditions on datasets not current state
+- status on datasets not aligned with project status
+- typeOfData on datasets is a weird list of options
+- does dataset need all the date fields?
```
Comment (mpro7) [inline docs/data/datamodel.md:333]:
```suggestion
- access conditions on datasets not current state,
- status on datasets not aligned with project status,
- typeOfData on datasets is a weird list of options,
- does dataset need all the date fields?
```

**Review by mpro7 — APPROVED** [review]

File: docs/data/datamodel.md (hunk 8 above)
Comment (BalduinLandolt) [reply docs/data/datamodel.md:143]: Thanks. I'm going for `#### String with Language Tag (lang_string)` instead though, to make it clear that it's not _only_ the tag, but the string with the tag.

File: docs/data/datamodel.md (hunk 10 above)
Comment (BalduinLandolt) [reply docs/data/datamodel.md:165]: meant to write "like a PID", thanks for spotting.

File: docs/data/datamodel.md (hunk 11 above)
Comment (BalduinLandolt) [reply docs/data/datamodel.md:333]: I actually meant to remove this entirely, before merging. Thanks, would otherwise have forgotten about this.

**Pull request merged**